Avoid warning messages when a publisher unregisters

lucasw opened the pull request:
This doesn't fix the underlying problem of unregister from #111, just hides it.

I assume that unregistering a publisher works sufficiently well that it is safe to do (and re-registering and unregistering any amount of times), that the only downside to unregistering beyond the warning message is the lingering memory and processing overhead that leads to the warning message- clearly state is not returned to the same as where the publisher never existed in the first place.

Maybe some additional logic could be introduced that could distinguish between a 'real' TransportInitError (caused by network issues?) and the unregister case?  The `if not rospy.core.is_shutdown_requested():` line  (https://github.com/ros/ros_comm/blob/melodic-devel/clients/rospy/src/rospy/impl/tcpros_base.py#L342-L347) is already handling node shutdown differently, maybe something else could happen if unregistration can be detected.

dirk-thomas committed:
update test to pass with old and new yaml (#1893)

dirk-thomas committed:
fix flakyness of transform test (#1890)

dirk-thomas committed:
fix dictionary changed size during iteration (#1894)

dirk-thomas committed:
fix flakyness of test_topic_statistics (#1896)

* add assertion to narrow down error case

* update test to ignore messages which are insufficient

dirk-thomas committed:
increase time limit of advertisetest/publishtest.test to reduce flakyness (#1897)

* increase time limit of publishtest.test to reduce flakyness

* increase time limit of advertisetest.test to reduce flakyness

dirk-thomas committed:
make statistics.test less flaky (#1899)

mm318 committed:
fix AttributeError: __exit__ (#1915)

Signed-off-by: Miaofei <miaofei@amazon.com>

smnogar committed:
fix mac trying to use epoll instead of kqueue (#1907)

macOS was incorrectly attempting to use epoll/poll instead of kqueue, occassionally generating errors such as RuntimeError: concurrent poll() invocation and breaking python nodes.

Co-authored-by: Steve Nogar <snogar@gmail.com>

smnogar committed:
add exception for ConnectionAbortedError (#1908)

Prevent python nodes from spamming the terminal every time a node is quit. Noted in Python 3 running on macOS.

Co-authored-by: Steve Nogar <snogar@gmail.com>

dirk-thomas committed:
Merge pull request #1917 from ros/backports_melodic

changes between 1.15.4 (Noetic) and 1.14.4 for backporting into Melodic

dirk-thomas committed:
update changelogs

dirk-thomas committed:
1.14.5

elektrokokke committed:
Fix brief description comments after members (#1920)

Current description comments after members start with '//!', resulting in broken documentation showing the descriptions on the wrong members. Change that to '//!<' according to doxygen documentation to have descriptions correctly associated with the respective members.

zbynekwinkler committed:
Remove extra \n in ROS_DEBUG. (#1925)

lucasw committed:
Avoiding warning messages when a publisher unregisters.  This doesn't fix the underlying problem of unregister not completely working, just hides it.

fujitatomoya committed:
Check if enough FDs are free, instead counting the total free FDs. (#1929)

Signed-off-by: Tomoya.Fujita <tomoya.fujita825@gmail.com>

Tobias-Fischer committed:
Fix #923 (use undefined dynamic_lookup on OSX) (#1923)

See https://github.com/conda-forge/ros-roslz4-feedstock/pull/3 and https://gitter.im/RoboStack/Lobby?at=5e830ffdbd09254f83ea7d4c and https://blog.tim-smith.us/2015/09/python-extension-modules-os-x/ for details.

jspricke committed:
[roslaunch] remove pycrypto import (not used) (#1922)

pycrypto used to be installed through rosbag. This was dropped in #1609
making roslaunch on remote hosts fail.

dirk-thomas commented:
@ros-pull-request-builder retest this please

lucasw committed:
Sort printed nodes by namespace alphabetically (#1934)

dirk-thomas commented:
> Maybe some additional logic could be introduced that could distinguish between a 'real' TransportInitError (caused by network issues?) and the unregister case?

Without that separation I don't think it is a good idea to demote the message from `warning` to `debug`.

sloretz committed:
[noetic] Avoid infinite recursion in rosrun tab completion when rosbash is not installed (#1948)

* Avoid infinite recursion

This avoids infinite recusion that happens if tab completion is
attempted but rosbash is not installed.

Signed-off-by: Shane Loretz<sloretz@openrobotics.org>
Signed-off-by: Shane Loretz <sloretz@osrfoundation.org>

* remove empty line

Co-authored-by: Dirk Thomas <dirk-thomas@users.noreply.github.com>

TSchildhauer committed:
Fix bag migration failures caused by typo in connection_header assignment (#1952)

Bag migration failed with follwoing errors:
NameError: global name 'connection_header' is not defined
This change fixes that by using local name conn_header

Barry-Xu-2018 committed:
Fix a bug that using a destroyed connection object. (#1950)

there is a case that accessing a connection object which was destroyed by
ConnectionManager and TransportSubscriberLink/TransportPublicationLink
while getting a close event in pollset to call Connection::drop.

Signed-off-by: Barry Xu <Barry.Xu@sony.com>

dirk-thomas committed:
fix NameError in launch error handling (#1965)

dirk-thomas committed:
Merge pull request #1959 from ros/backports_melodic

changes between 1.15.7 (Noetic) and 1.14.5 for backporting into Melodic

dirk-thomas committed:
update changelogs

dirk-thomas committed:
1.14.6

peci1 committed:
Added const versions of XmlRpcValue converting operators. (#1978)

efernandez committed:
Add latch param to throttle (#1944)

This allows to force the advertised throttled publisher to be latched or
not, instead of using the same latching property value as the input
topic.

It defaults to the previous behaviour, i.e. it uses the same latching
property value as the input topic.

cjds committed:
Add which node has been registered with the same name (#1992)

* Add which node has been registered with the same name

Was looking through logs of several nodes that had been launched at the same time and it was getting hard to tell which node was being duplicated. This would allow you do to see the name of the node that caused the issue

* switch to shutdown_node_task

dirk-thomas committed:
remove not existing NodeProxy from rospy __all__ (#2007)

dirk-thomas committed:
Merge pull request #2008 from ros/backports_melodic

changes between 1.15.8 (Noetic) and 1.14.6 for backporting into Melodic

dirk-thomas commented:
@lucasw This seems to need a rebase / resolve conflicts to be mergable and not show unrelated commits.

dirk-thomas commented:
I will go ahead and close this for now due to no further responses. Please feel free to comment on the closed ticket once the feedback has been addressed and the ticket can be reopened.
